src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,41 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    public static final int MAX_VALUE = 100;
+    private final Random random;
+    private final ColorSupplier colorSupplier;
+    private Shape figure;
+
+    public FigureSupplier() {
+        random = new Random();
+        colorSupplier = new ColorSupplier();
+    }
+
+    public Figure getRandomFigure() {
+        String color = colorSupplier.getRandomColor();
+        int index = random.nextInt(Shape.values().length);
+        figure = Shape.values()[index];
+        double side1 = random.nextDouble() * MAX_VALUE;
+        double side2 = random.nextDouble() * MAX_VALUE;
+        double side3 = random.nextDouble() * MAX_VALUE;
```

Comment:
lets implement DRY principle
```suggestion
        double defaultSide = random.nextDouble() * MAX_VALUE;
```